editoast: fix infra update

Erashin opened the pull request:
Remove default values in infra model (except modified_date which must be always modified) and replace them with Option. Set default values at creation. Default values will not be set at update anymore.

Fixes #4108.

flomonster commented on editoast/openapi.yaml:
```diff
@@ -187,6 +187,13 @@ paths:
       tags:
         - infra
       summary: Create an infra
+      parameters:
+        - in: query
+          name: locked
+          description: whether or not the infra can be edited/deleted.
+          default: false
+          schema:
+            type: bool
```
We don't need to specify if an empty infra should be locked or not.

Erashin replied:
100%. Confusion with my other PR. Will get rid of this.

flomonster reviewed (changes requested):
Thanks for this bug fix. Except for the unnecessary query param, everything looks good to me.

Erashin committed:
editoast: fix infra update

Remove default values in infra model (except modified_date which must be always modified) and replace them with Option. Set default values at creation. Default values will not be set at update anymore.

clarani reviewed (commented):
LGTM

flomonster reviewed (approved):
LGTM. Thanks for this bug fix :tada:
